onramps style guide (#12981)
newrelic/docs-website: src/content/docs/style-guide/writing-docs/article-templates/onramps-style-guide.mdx

---
title: Onramps style guide
contentType: page
template: basicDoc
topics:
  - onramps
---

import apmOnrampsIssuesExample from 'images/apm_screenshot-crop_onramps-issues-example.webp'

import apmEmptyStateExample from 'images/apm_screenshot-crop_empty-state-example.webp'

import apmActiveStateExample from 'images/apm_screenshot-crop_active-state-example.webp'

import apmServiceLevelsExample from 'images/apm_screenshot-crop_service-levels-example.webp'

Our onramps content is the help text that appears in the UI when you click the **?** icon. The purpose of in-product help is to help users understand the UI, interpret performance data, and New Relic solutions with their system problems. Onramps provide more detail than a [tooltip](https://one-core.newrelic.com/components/tooltip/). Ideally, they give users essential info so they can stay in the product and keep working, rather than switching over to read the docs. When writing onramps, include information that helps users:

> - Understand what a feature is used for, and the benefits of the feature beyond the existing heading or other content. 
> - Perform specific tasks.
> - Understand what to do next. Are there recommended follow-on actions? The Apdex troubleshooting onramp provides a nice example for this. It tells users: "Compare the Apdex chart with other charts (like Web transaction time and Error rate) to look for the causes for the Apdex change."

Author: ```suggestion
* Understand what a feature is used for, and the benefits of the feature beyond the existing heading or other content. 
* Perform specific tasks.
* Understand what to do next. Are there recommended follow-on actions? The Apdex troubleshooting onramp provides a nice example for this. It tells users: "Compare the Apdex chart with other charts (like Web transaction time and Error rate) to look for the causes for the Apdex change."
```

---
title: Onramps style guide
contentType: page
template: basicDoc
topics:
  - onramps
---

import apmApmSummaryOnrampsGif from 'images/apm_screenshot-full_apm-summary-onramps-gif.gif'

import apmOnrampsIssuesExample from 'images/apm_screenshot-crop_onramps-issues-example.webp'

import apmEmptyStateExample from 'images/apm_screenshot-crop_empty-state-example.webp'

import apmActiveStateExample from 'images/apm_screenshot-crop_active-state-example.webp'

import apmServiceLevelsExample from 'images/apm_screenshot-crop_service-levels-example.webp'

> Our onramps content is the help text that appears in the UI when you click the **?** icon. The purpose of in-product help is to help users understand the UI, interpret performance data, and New Relic solutions with their system problems. Onramps provide more detail than a [tooltip](https://one-core.newrelic.com/components/tooltip/). Ideally, they give users essential info so they can stay in the product and keep working, rather than switching over to read the docs. 

Author: ```suggestion
Our in-product help content provides users with the tools they need to understand the New Relic UI, interpret performance data, and learn how new relic solutions can help with their problems. Onramps provide more detail than a [tooltip](https://one-core.newrelic.com/components/tooltip/). Ideally, they give users essential info so they can stay in the product and keep working, rather than switching over to read the docs. 
```

<img

>     title=""

Reviewer: Missing title and alt text (although not a dealbreaker on an internal-only doc like this)

> <img
>     title=""
>     alt=""
>     src={apmApmSummaryOnrampsGif}
> />

Author: ```suggestion
<img
    title="Onramps example"
    alt="gif of onramps example in the UI"
    src={apmApmSummaryOnrampsGif}
/>
```

> ## Content guidelines

Reviewer: To me the entire doc reads like "Content guidelines." This section is about the heart of the content and what makes it really sing, whereas the rest of the doc is more like rules. I'd shoot for a heading that acknowledges that. What about something like, "What does a great on-ramp look like" or "Writing a great on-ramp" or similar.

> * Be clear and concise. Contextual help should be clear and concise. Avoid using jargon or technical terms that users may not understand.

Reviewer: Here and subsequently, I would bold the main idea to make these bullets easier to scan. (`**Be clear and concise.**`)

* Use second-person point of view to address the reader. Address the user as "you", such as "We couldn't find your data." 
* Answer the "why" and "how" questions. For example, a user might ask themselves, "Why is this important to me?" or "How do I use this information to improve my app performance?" Write content to help answer these questions, to understand the larger context of their application. For example, a user can learn about how APM and logs relate to each other, or how APM and infrastructure relate to each other.
* Provide additional context to clarify ambiguous or confusing areas in the product. For example, if there is something that is not clearly defined in the UI, we can provide additional information to help users understand it.

> * Help users get more out of New Relic (and increase our revenue). When users are able to use New Relic more effectively, they are more likely to renew their subscriptions and recommend us to others. This, in turn, helps us to make more money. Include information that helps users:

Reviewer: I think we talked about this in the sprint review, but let's reframe these bullets a bit

  * Understand what a feature is used for, and the benefits of the feature beyond the existing heading or other content. 
  * Perform specific tasks.
  * Understand what to do next. Are there recommended follow-on actions? The Apdex troubleshooting onramp provides a nice example for this. It tells users: "Compare the Apdex chart with other charts (like Web transaction time and Error rate) to look for the causes for the Apdex change."

> ## FAQs

Reviewer: I'd like us to not use the word FAQ in our docs, including in style guide.

In terms of restructuring, I think you have two types of content a bit mixed under this FAQ:

* Rules-type content:bold/italics
    * On-ramp titles
    * Length
    * Bold/italics
    * Links
    * Tile states
    * UI navigation info
* More philosophical info, which may actually belong above with the other philosphical info
   * When is onramps contgent needed?
   * Level of detail

A few ideas for how to restructure it:

* Just lose the FAQ header and make the questions all h2s
* Move the philosphical ones up into the earlier section, and group the others together here under a Rules-type heading
* Move all of them into a table to make the structure a bit more skimmable

  * **Empty state**: If the tile has no data, it's an empty state, and if the user clicks on it, they may be directed to a setup page. 
    <img
    title=""
    alt=""
    src={apmEmptyStateExample}
    style={{ maxWidth: '75%' }}
/>

  * **Active state**: If the has data, it's an active state, and our onramps content will want to help users interpret the data.
    <img
    title=""
    alt=""
    src={apmActiveStateExample}
    style={{ maxWidth: '75%' }}
/>

In the examples above, regardless of what state the tile is in, the onramp is static, so it's important to contain information that's helpful for both states. Here's the onramp content for the **Service levels** tile.

 <img
    title=""
    alt=""
    src={apmServiceLevelsExample}
    style={{ maxWidth: '75%' }}
/>

### How long should the content be?

If you're not using collapsers, **keep the character count under 1200** for the entire onramps section.

### Should I bold/italicize to emphasize something?

Since we can't add code formatting to onramps content, use bold and italics to emphasize something as needed.

### How much detail should I include?

> If the onramps content isn't in the main place where the feature is (i.e., onramps content for the **Service levels** tile on the **APM Summary** page), then it’s not the place to include a lot of detail.

Reviewer: Avoid `i.e.` and `e.g.` (a few instances of `i.e.` in this doc)

https://docs.newrelic.com/docs/style-guide/word-choice/usage-dictionary/#eg

### Should I add links to docs?

It's okay to add a link to docs for more details on something, but here are some considerations:
* Add a docs link if there's relevant information and it's too long to summarize/explain in onramps. 
* The onramps content will persist after the user clicks on something, so add a docs link if it's relevant to where the user might go next (i.e., if you click on the **Issues** tile and are taken to the **Issues** page, you'll still see the onramps popup on the same page, so you may want to learn how to configure issue thresholds on this page).

Avoid making a link to docs the sole content of an onramp.

> When linking, use the established UI style for docs links. Use "See our docs" for the text.

Reviewer: Generally we use code backticks rather than quotes in the style guide, although I don't think this is written down. See usage dictionary for some examples

---
title: Onramps style guide
contentType: page
template: basicDoc
topics:
  - onramps
---

import apmOnrampsIssuesExample from 'images/apm_screenshot-crop_onramps-issues-example.webp'

import apmActiveEmptyState from 'images/apm_screenshot-crop_active-empty-state.webp'

> 

Reviewer 2: @ally-sassman It would be great to add a section in this style guide doc about the purposes of on-ramps content:
1. We want to enable users to get more value out of the UI
2. We want to help provide a larger context for why and how people can use these pieces of the UI to solve their problems and connect them to other parts of the UI
3. We want to provide info on how users can solve specific problems with the part of the UI they're looking at.

Reviewer 2: Feel free to chat with me about this if you want to talk through it.

Author: **Notes**

The purpose is to help people with the thing they're looking at when it's not obvious.

The most important questions are the why and the how.

However, we want the UI to speak for itself, so we only use it when there's ambiguity. 

Users also use it to solve a problem (i.e., why is the Apdex page important to me?) 

There's a larger context - how does a piece of the UI fit into the larger context? I.e., how do APM and logs relate to each other? How does APM and infrastructure relate to either other? Here's how this aspect of APM relates to this piece of infrastructure. How are Issues related to Alerts? 

The secondary goal is if there's ambiguity/something's not clearly defined, we can take action as the docs team to provide a patchwork of sorts, filling in where there may be confusion.

The business goal is that in-product help will help users get more out of NR, and in turn we make more money.

Onramps content is information that appears our UI when a user clicks the **?** icon. This content is meant to be truncated and contextualized to where the user is and what they need to know to perform a task.

> ## How should I name the section titles?

Reviewer 3: Think we should say that headers can be any format, as long as they're short/succinct. We have quite a few that are questions, and we have quite a few that are not questions, and maybe we specifically say there is no rule around that.

Reviewer 4: Recommendation for heading: make sure it's clear what the heading relates to. For example, don't say, "What is this?" -- instead say "What's an issue?" This is important because the info is off the side of the page, and if you change the page the onramps stay open.

When applicable, use an FAQ format where the headers are in question format.

> For example:

Author: ```suggestion
For example, here's the Issues tile with headers formatted into questions.
```

> For example, here's the Issues tile with headers formatted into questions.

Reviewer 5: Should issues be bolded? I'm not actually sure myself so I figure it can't hurt to ask!

<img
    title=""
    alt=""
    src={apmOnrampsIssuesExample}
    style={{ maxWidth: '50%' }}
/>

If FAQ format is not applicable, use our standard headers for titles.

## What if a tile has two states (active and empty)?

> A tile is something that's clickable and may or may not contain data, depending on if customers have set it up. This is called an active vs. an empty state, and it's something to consider when writing onramps content. If the tile has no data, it's in empty state, and if the user clicks on it, they may be directed to a set up page. If it has data, it's in active state, and our onramps content will want to help users interpret the data.

Reviewer 5: I always found "or may not" to be covered by "may", but it's not a big deal.

> A tile is something that's clickable and may or may not contain data, depending on if customers have set it up. This is called an active vs. an empty state, and it's something to consider when writing onramps content. If the tile has no data, it's in empty state, and if the user clicks on it, they may be directed to a set up page. If it has data, it's in active state, and our onramps content will want to help users interpret the data.

Reviewer 5: Should there be some differentiator for "active" and "empty" state?

For example, here's the Service levels tile in active and empty states. The onramps content should contain info that's helpful for both states.

<img
    title=""
    alt=""
    src={apmActiveEmptyState}
    style={{ maxWidth: '75%' }}
/>

## How long should the content be?

> If you're not using collapsers, **keep the character count under 1200** for the entire onramps section.

Reviewer 3: did we want to add a rule for limiting a sub-section of an onramp?

Author: @paperclypse What do you think? My thought against adding it is it feels subjective to what the onramps is about. What if the onramp only needs one or two sections (vs. 3-5)? Then the sections could be longer if there are only 1-2, and s section limit wouldn't always be applicable to all onramps.

Reviewer 2: I hesitate to add hard and fast rules like this... But maybe we should rephrase this as: Abide by these limits, unless you have a really good reason not to.

> ## How should I bold/italicize to emphasize something?

Reviewer 5: I think you can cut the "how" here? It seems that the subsection is covering whether or not you should bold/italicize for emphasis, not how to actually do it.

Since we can't add code formatting to onramps content, use bold and italics to emphasize something as needed.

## How much detail should I include?

> If the onramps content isn't in the main place where the feature is (i.e., onramps content for the Service levels tile on the APM Summary page), then it’s not the place to include a lot of detail.

Reviewer 5: I think APM Summary should be bolded.

> ## Should I add links out to docs?

Reviewer 5: Links out is makes my brain stop for a second. Maybe "Should I link to other docs?" might be easier to parse? Just a thought.

It's okay to add a link to docs "for more details" on something, but here are some considerations:
* Add a docs link if there's relevant information and it's too long to summarize/explain in onramps. 

> * The onramps content will persist after the user clicks on something, so add a docs link if it's relevant to where the user might go next (i.e., if you click on the Issues tile and are taken to the **Issues** page, you'll still see the onramps popup on the same, so you may want to learn how to configure issue thresholds on this page).

Reviewer 4: ```suggestion
* The onramps content will persist after the user clicks on something, so add a docs link if it's relevant to where the user might go next (i.e., if you click on the Issues tile and are taken to the **Issues** page, you'll still see the onramps popup on the same page, so you may want to learn how to configure issue thresholds on this page).

Avoid making a link to docs the sole content of an onramp.  When linking, use the established UI style for docs links. Text is "See our docs" and use the external link icon. 
```

Reviewer 3: regarding Michelle's comment above, the external link icon she mentions works automatically for links so we don't have to mention that apparently.

## How should I reference UI navigation instructions?

>  Let's say you want to direct users to something relevant in the UI and need to add navigation instructions. For example, if a user clicks on the Issues tile and they want to get alerts for issues (which is on another UI page), you can provide instructions in two ways:

Reviewer 5: Like elsewhere, should Issues be bolded?

>  * Link them to the docs (i.e., "To create an alert condition, see [Create your first alert](https://docs.newrelic.com/docs/alerts-applied-intelligence/new-relic-alerts/get-started/your-first-nrql-condition/).")

Reviewer 5: I think the link formatting is wrong here with the https: part.

Reviewer 4: This one takes them out of the product, so I wouldn't put it first. And is it using a link to docs as shorthand for more info? That's why I recommend that a link to docs not be the only content in an onramp.

Reviewer 3: and reminder about Michelle's point; we apparently want to follow the UI writing style of only using 'see our docs' as the only link text.

---
title: Onramps style guide
contentType: page
template: basicDoc
topics:
  - onramps
---

import apmOnrampsIcon from 'images/apm_icon_onramps-icon.webp'

import apmActiveEmptyState from 'images/apm_screenshot-crop_active-empty-state.webp'

Onramps content is information that appears our UI when a user clicks a <img
    title=""
    alt=""
    src={apmOnrampsIcon}
/>
 icon. This content is meant to be truncated and contextualized to where the user is and what they need to know to perform a task.

## How should I name the titles of onramps content?

### FAQ structure

When applicable, use an FAQ format where the headers are in question format.

Example:

**Can I get notifications for issues?**
You can get notifications through email, Slack, Jira, and more. To learn how to set up notifications, see [Introduction to alert notifications](/docs/alerts-applied-intelligence/notifications/intro-notifications/).

### Regular headers

If FAQ format is not applicable, use our standard headers for titles.

Example:

**Forward logs to New Relic**

> Forwarding logs to New Relic allows you to have all logs in one place.

Author: @zuluecho9 Do you have any examples of onramps that's not in a question format? I just added this as placeholder.

## What if a tile has two states (active and empty)?

A tile is something that's clickable and may or may not contain data, depending on if customers have set it up. This is called an active vs. an empty state, and it's something to consider when writing onramps content. If the tile has no data, it's in empty state, and if the user clicks on it, they may be directed to a set up page. If it has data, it's in active state, and our onramps content will want to help users interpret the data.

> For example, here's the Incidents tile in active and empty states, and the content that touches on both scenarios.

Author: ```suggestion
For example, here's a tile in active and empty states, and the content that touches on both scenarios.
```
